module/VuFind/tests/unit-tests/src/VuFindTest/Search/Base/ParamsTest.php

```diff
@@ -149,13 +152,42 @@ public function testGetFacetLabel(): void
         $this->assertEquals('foo_label', $params->getFacetLabel('foo'));
 
         // If we add a checkbox facet for a field that already has an assigned label,
-        // we do not expect the checkbox label to override the field label:
+        // we expect the checkbox label to override the field label:
         $params->addCheckboxFacet('foo:bar', 'checkbox_label');
-        $this->assertEquals('foo_label', $params->getFacetLabel('foo', 'bar'));
+        $this->assertEquals('checkbox_label', $params->getFacetLabel('foo', 'bar'));
         $this->assertEquals('foo_label', $params->getFacetLabel('foo', 'baz'));
         $this->assertEquals('foo_label', $params->getFacetLabel('foo'));
     }
 
+    /**
+     * Test that getFacetLabel works as expected with aliases.
+     *
+     * @return void
+     */
+    public function testGetFacetLabelWithAliases(): void
+    {
+        $params = $this->getMockParams();
+        $this->setProperty(
```

Comment:
I wonder if this is a sign that we need an addFacetAlias() method, so it's not necessary to rely on reflection here... but that's probably a matter for a separate PR.